TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/ReplayEngine.java

```diff
@@ -129,6 +131,11 @@ public boolean isWorkOutstanding() {
         var start = timeShifter.transformSourceTimeToRealTime(originalStart);
         var end = timeShifter.transformSourceTimeToRealTime(originalEnd);
         var interval = numPackets > 1 ? Duration.between(start, end).dividedBy(numPackets-1) : Duration.ZERO;
+        metricsLogger.atSuccess()
+                .addKeyValue("requestId", requestKey.toString())
+                .addKeyValue("delayFromOriginalToScheduledStartInMs", Duration.between(originalStart, start).toMillis())
```

Comment:
This is good for now, but I'm also hoping to see a metric for drift between actual and scheduled.

Reply:
Yes! I'd like to include that as well, but didn't see an obvious place to put it in. Do you have a suggestion for that?

Reply:
RequestSenderOrchestrator currently has 4 uses of EventLoop::scheduleEvent().  That returns a ScheduledFuture.

You could add a helper function to coreUtils that takes an EventLoop, the arguments to scheduleEvent, and some metric data then wrap it all together to do the scheduling.  In the returned future, you can chain some metrics emission code to show the drift between scheduled and actual times.  This could/should be done with the RequestSenderOrchestrator having little to no knowledge that its schedules are being instrumented.  However, it should also make that call with a label or tag so that the implementation can determine what sensitivity to treat the metric emission at.

One thing to be aware of is that we schedule on a packet by packet basis, based upon the originally observed start-end times.  Those two bookend times might be things that are more critical than the intra-"packet" times.  However, having the internal datapoints will paint more complete pictures when diagnosing skew issues at a more macro level.  I just want to code to be flexible enough that it doesn't become overly expensive or inefficient past the point where it provides some users value.